src/audio/volume/volume.c

```diff
@@ -212,9 +220,12 @@ static struct comp_dev *volume_new(struct sof_ipc_comp *comp)
 	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
 		cd->volume[i]  =  MAX(MIN(cd->vol_max, VOL_ZERO_DB),
 				      cd->vol_min);
-		cd->tvolume[i] =  cd->volume[i];
+		cd->tvolume[i] = cd->volume[i];
+		cd->mvolume[i] = cd->volume[i];
+		cd->muted[i] = false;
 	}
 
+	cd->vol_ramp_active = false;
```

Comment:
\<rant\> I find we use zero-initialising allocations (rzalloc()) exceedingly, sometimes also where they aren't really required, and then we also additionally initialise fields of the just allocated object with 0 or NULL of false... Seems strange to me \</rant\>

Reply:
@lyakh I'd find it safer than forgetting to zero-initialize where it's needed, for one.

Reply:
I'll do whatever you recommend. I'm myself preferring to be explicit with things that need to be initialized. Let me know your preference.